src/doLifeCycle.js

```diff
@@ -0,0 +1,84 @@
+class LifeCycle {
+
+  transform = value => value;
+  beforeUpdate = [];
+  afterUpdate = [];
+
+  constructor(main, nextValue, opts = {}) {
+    this.main = main;
+    this.nextValue = nextValue;
+
+    this.ensureArgs(opts);
+  }
+
+  ensureArgs = opts => {
+    if (typeof opts === 'function') {
+      this.transform = opts;
+    } else if (typeof opts === 'object') {
+      if (opts.beforeUpdate || opts.afterUpdate) {
+        if (opts.transform) this.transform = opts.transform;
+        this.beforeUpdate = this.ensureArray(opts.beforeUpdate);
+        this.afterUpdate = this.ensureArray(opts.afterUpdate);
+      }
+    } else throw new Error('Illegal options supplied to cycle.');
+  }
+
+  ensureArray = item => [].concat(item);
+
+  ensureUpdatesAreFuncs = () => {
+    const { beforeUpdate, afterUpdate } = this;
+    if (
+      !beforeUpdate.every(func => typeof func === 'function') ||
```

Comment:
I'm still uncertain if we want to be responsible for all this validation. 

I feel like by saying, "you should pass function(s)", it's on the user to get it right. Type checking (I feel) is beyond the scope of js, unless we are deliberately executing some logic based on the type of something.

I think this would be better suited for typescript.